Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -284,14 +288,18 @@ final class PurchasesOrchestrator {
                                      discountIdentifier: discountIdentifier,
                                      product: product,
                                      subscriptionGroupIdentifier: subscriptionGroupIdentifier) { result in
-                completion(result)
+                self.operationDispatcher.dispatchOnMainActor {
+                    completion(result)
+                }
             }
         } else {
                 self.sk1PromotionalOffer(forProductDiscount: productDiscount,
                                          discountIdentifier: discountIdentifier,
                                          product: product,
                                          subscriptionGroupIdentifier: subscriptionGroupIdentifier) { result in
-                    completion(result)
+                    self.operationDispatcher.dispatchOnMainActor {
+                        completion(result)
+                    }
```

Comment:
We're dispatching on main actor for some and on main thread on the other, is there a difference / preference?

Reply:
not sure there are any guarantees for order of operation between operation added from queue vs code in actor but I'd imagine not

Reply:
No reason to use GCD methods anymore, converted them to actor.

I guess technically there is no difference since we're just dispatching to the main thread in both cases.

Reply:
Longer term that's the way to go: get completely rid of all the GCD stuff, switch to modern Swift Concurrency, and also adopt the Swift 6 concurrency model.